forch/port_state_manager.py

```diff
@@ -22,63 +21,93 @@ def register(func):
 
 
 class PortStateMachine:
+
     """State machine class that manages testing states of an access port"""
 
-    UNAUTHENTICATED = 'unauthenticated'
-    AUTHENTICATED = 'authenticated'
-    SEQUESTERED = 'sequestered'
-    OPERATIONAL = 'operational'
-    INFRACTED = 'infracted'
+    UNAUTHENTICATED = DVAState.State.unauthenticated
+    SEQUESTERED = DVAState.State.sequestered
+    OPERATIONAL = DVAState.State.operational
+    INFRACTED = DVAState.State.infracted
 
-    TRANSITIONS = {
+    _TRANSITIONS = {
         UNAUTHENTICATED: {
-            PortBehavior.cleared: OPERATIONAL,
-            PortBehavior.sequestered: SEQUESTERED,
+            PortBehavior.Behavior: {
+                PortBehavior.Behavior.cleared: OPERATIONAL,
+                PortBehavior.Behavior.sequestered: SEQUESTERED,
+            }
         },
         SEQUESTERED: {
-            PortBehavior.passed: OPERATIONAL,
-            PortBehavior.failed: INFRACTED,
-            PortBehavior.deauthenticated: UNAUTHENTICATED,
+            PortBehavior.Behavior: {
+                PortBehavior.Behavior.deauthenticated: UNAUTHENTICATED
+            },
+            SessionResult.ResultCode: {
+                SessionResult.ResultCode.PASSED: OPERATIONAL,
+                SessionResult.ResultCode.FAILED: INFRACTED,
+            }
         },
         OPERATIONAL: {
-            PortBehavior.cleared: OPERATIONAL,
-            PortBehavior.deauthenticated: UNAUTHENTICATED,
+            PortBehavior.Behavior: {
+                PortBehavior.Behavior.cleared: OPERATIONAL,
+                PortBehavior.Behavior.deauthenticated: UNAUTHENTICATED
+            }
         },
     }
 
+    class StateCallbacks:
+        """Wrapper for state machine callbacks"""
+        unauthenticated_state = None
```

Comment:
seems odd to have a whole class for this rather than just a map indexed by the relevant constants.... not a big deal but this seems unnecessarily verbose.